src/Drivers/Database.php

```diff
@@ -23,10 +23,26 @@ public function audit(Auditable $model): ?Audit
     public function prune(Auditable $model): bool
     {
         if (($threshold = $model->getAuditThreshold()) > 0) {
-            return $model->audits()
+            $class = get_class($model->audits()->getModel());
+            $auditKeyName = (new $class)->getKeyName();
+
+            $forRemoval = $model->audits()
                 ->latest()
-                ->offset($threshold)->limit(PHP_INT_MAX)
-                ->delete() > 0;
+                ->get()
+                ->slice($threshold)
+                ->pluck($auditKeyName);
+
+            if (!$forRemoval->isEmpty()) {
+                $forRemovalChunks = $forRemoval->chunk(Config::get('audit.placeholders_limit', 50000));
+                $answer = false;
+                foreach ($forRemovalChunks as $chunk) {
+                    $answer = $answer || $model->audits()
```

Comment:
Maybe I don't understand this,
after the first `chunk`, `answer` would be `true`,
then in the second chunk(and later), the delete would no longer be executed

Reply:
Also, if the first chunk is executed, and the second for some reason is not, would it return `true`/`false` even though only a portion of the records were deleted?

Reply:
any ideas? I thought if some deletion was  performed in any case return true